internals/plan/plan.go

```diff
@@ -643,6 +705,30 @@ func CombineLayers(layers ...*Layer) (*Layer, error) {
 		}
 	}
 
+	// Combine the same sections from each layer.
+	for field, extension := range layerExtensions {
+		var sections []LayerSection
+		for _, layer := range layers {
+			if section := layer.Section(field); section != nil {
+				sections = append(sections, section)
+			}
+		}
+		// Deliberately do not expose the zero section condition to the extension.
+		// For now, the result of combining nothing must result in an omitted section.
+		if len(sections) > 0 {
```

Comment:
Not sure why this is checked here rather than inside `extension.CombineSections`

Reply:
As the comment explains, I do not want to give the implementer the option to return something, even if no sections are provided as input. For now, I want to restrict this, and prevent section implementations (i.e. netstate) from making a call how to behave in this case.